src/saml2/saml.py

```diff
@@ -205,9 +205,11 @@ def set_text(self, val, base64encode=False):
             val = _b64_encode_fn(val)
             self.set_type("xs:base64Binary")
         else:
-            if isinstance(val, six.binary_type):
+            if typ == "xs:anyType":
+                pass
+            elif isinstance(val, six.binary_type):
                 val = val.decode()
-            if isinstance(val, six.string_types):
+            elif isinstance(val, six.string_types):
```

Comment:
Changing this to `elif` is not correct. When the value is of `binary_type` it is `decode()`d to become a `string_type`, and handling continues from there.

Reply:
This hints that more tests are needed to cover all the types.

Reply:
Based on #539 , this modified code works:
```
diff --git a/src/saml2/saml.py b/src/saml2/saml.py
index 023f352..df23c92 100644
--- a/src/saml2/saml.py
+++ b/src/saml2/saml.py
@@ -223,6 +223,8 @@ class AttributeValueBase(SamlBase):
                             _ = float(val)
                         elif typ == "xs:base64Binary":
                             pass
+                        elif typ == "xs:anyType":
+                            pass
                         else:
                             raise ValueError("Type and value doesn't match")
             elif isinstance(val, bool):
```

I did not submit a pull request because it sounds like @c00kiemon5ter wants to refactor this function anyway.